### PR TITLE
add BFloat16 support for fold and unfold on CPU

### PR DESCRIPTION
**Pull request by CaoE**
Add BFloat16 support for fold and unfold operators on CPU.

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -6682,7 +6682,7 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
            autodiff_nonfusible_nodes=["aten::hardswish"]),
     OpInfo('nn.functional.unfold',
            aten_name='im2col',
-           dtypes=floating_types_and(torch.half),
+           dtypes=floating_types_and(torch.half, torch.bfloat16),
```
Comment (zou3519) [inline torch/testing/_internal/common_methods_invocations.py:6685]: You'll want to separate this into dtypesIfCPU and dtypesIfCUDA: https://github.com/pytorch/pytorch/blob/9573e7a644544fbd623e652766d47c3e84d588de/torch/testing/_internal/common_methods_invocations.py#L4860-L4861

**Review by zou3519 — COMMENTED** [review]
This looks pretty good! Some tests are failing but I think we just need to modify the OpInfo to get this to work

File: torch/testing/_internal/common_methods_invocations.py (hunk 1 above)
Comment (CaoE) [reply torch/testing/_internal/common_methods_invocations.py]: I separated this into dtypesIfCPU. Thank you.

**Review by zou3519 — APPROVED** [review]
LGTM, thank you!

**Pull request closed without merging**